File: runtime/src/main/java/com/bennyhuo/activitybuilder/runtime/annotations/GenerateBuilder.java
package com.bennyhuo.activitybuilder.runtime.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by benny on 1/29/18.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface GenerateBuilder {
    GenerateMode mode() default GenerateMode.Both;
    boolean forResult() default false;
    ResultEntity[] resultTypes() default {};
}

File: annotations/src/main/java/com/bennyhuo/annotations/GenerateBuilder.kt
package com.bennyhuo.annotations

import kotlin.annotation.AnnotationRetention.BINARY
import kotlin.annotation.AnnotationTarget.PROPERTY

/**
 * Created by benny on 1/29/18.
 */
@Target(AnnotationTarget.CLASS)
@Retention(BINARY)
annotation class GenerateBuilder

@Target(PROPERTY)
@Retention(BINARY)
annotation class Required(val name: String)